### PR TITLE
Test for transposing a Square SGSparseMatrix .

### PR DESCRIPTION
**Pull request by sudk1896**
This adds a test for Transposing and checking the sanity for a `Square SGSparseMatrix`.
#2748 
@karlnapf @iglesias Review.
Meanwhile, you can suggest more tests that you'd like to see.

### CODE REVIEW

File: tests/unit/lib/SGSparseMatrix_unittest.cc [hunk 1]
```diff
@@ -265,3 +265,25 @@ TEST(SGSparseMatrix, from_dense)
 		for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
 			EXPECT_EQ(sparseMatrix(featIndex,vecIndex), denseMatrix(featIndex,vecIndex));
 }
+
+TEST(SGSparseMatrix, test_transposed_square_matrix)
```
Comment (karlnapf) [inline tests/unit/lib/SGSparseMatrix_unittest.cc]: dont need the "test_" prefix in the name

File: tests/unit/lib/SGSparseMatrix_unittest.cc [hunk 2]
```diff
@@ -265,3 +265,25 @@ TEST(SGSparseMatrix, from_dense)
...
+{
+	//Test for transposing Sparse Matrix.
```
Comment (karlnapf) [inline tests/unit/lib/SGSparseMatrix_unittest.cc]: superflous comment

File: tests/unit/lib/SGSparseMatrix_unittest.cc [hunk 3]
```diff
@@ -265,3 +265,25 @@ TEST(SGSparseMatrix, from_dense)
...
+	//number_of_features = number_of_vectors.
+	const float64_t sparseLevel=0.1;
```
Comment (karlnapf) [inline tests/unit/lib/SGSparseMatrix_unittest.cc]: we dont do camelCase in Shogun

File: tests/unit/lib/SGSparseMatrix_unittest.cc [hunk 4]
```diff
@@ -265,3 +265,25 @@ TEST(SGSparseMatrix, from_dense)
...
+	SGSparseMatrix<float64_t> sparseMatrix(numberOfFeatures, numberOfVectors);
+	GenerateMatrix<SGSparseMatrix<float64_t> >(sparseLevel, numberOfFeatures, numberOfVectors, randSeed, &sparseMatrix);
+
+	SGSparseMatrix<float64_t> sparseMatrixT=sparseMatrix.get_transposed();
+
+	EXPECT_EQ(sparseMatrix.num_features, sparseMatrixT.num_vectors);
+	EXPECT_EQ(sparseMatrix.num_vectors, sparseMatrixT.num_features);
+
+	for (index_t featIndex=0; featIndex<numberOfFeatures; ++featIndex)
+		for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
```
Comment (karlnapf) [inline tests/unit/lib/SGSparseMatrix_unittest.cc]: we use {} in loops and if then else if there is more than a line afterwards

**Comment by karlnapf** [comment]
This is a good test, did you check it with valgrind for memory leaks?
For that, just run it isolated (using the `gtest_filer` argument of the unit test), and see whether all allocated blocks are also freed

**Comment by karlnapf** [comment]
Few minor style changes needed, then we can merge. Thanks! :)

File: tests/unit/lib/SGSparseMatrix_unittest.cc (hunk 1 above)
Comment (sudk1896) [inline tests/unit/lib/SGSparseMatrix_unittest.cc]: @karlnapf Btw, all the other tests do not implement the style fixes you suggested (about camel case and the for loops). You can check.

**Comment by sudk1896** [comment]
@karlnapf : Btw, all the tests in `SGSparseMatrix` do not implement the style fixes you mention (Camel case and the braces).

**Comment by karlnapf** [comment]
Thats ok, leave the things as they are. Just for new ones we want to be consistent with you style.
If you feel inspired, feel free to send a style fix PR (low priority)

**Comment by karlnapf** [comment]
So in the added lines, there should be no camelcase anywhere.

The test names can be camel case (so exactly the opposite way of your update:) )

**Comment by sudk1896** [comment]
@karlnapf: Should I remove the braces then ? None of the tests except mine have them.

File: tests/unit/lib/SGSparseMatrix_unittest.cc [hunk 5]
```diff
@@ -265,3 +265,27 @@ TEST(SGSparseMatrix, from_dense)
 		for (index_t vecIndex=0; vecIndex<numberOfVectors; ++vecIndex)
 			EXPECT_EQ(sparseMatrix(featIndex,vecIndex), denseMatrix(featIndex,vecIndex));
 }
+
+TEST(SGSparseMatrix, transposed_square_matrix)
+{
+
+	const float64_t sparseLevel=0.1;
```
Comment (karlnapf) [inline tests/unit/lib/SGSparseMatrix_unittest.cc]: can you pls remove the camel case in variables?

**Comment by sudk1896** [comment]
@karlnapf: The file had a lot of camel case names (especiall for `sparseLevel` variable). I replaced all of them with `sparse_level`.

**Comment by karlnapf** [comment]
As I said above, please do not mix style fixes with adding new code in a single PR.

Patch 1: your new test, with the correct naming of variables
Pathc 2(optional): changing the style in the other tests in the file

**Comment by sudk1896** [comment]
@karlnapf: I haven't mixed the styles. **All** the camel case variables in this test file have been changed as you asked `variableName` to `variable_name`.

**Comment by karlnapf** [comment]
Yep but that's not what I wrote. Please don't change style in existing code in this patch, send a new one for such changes

**Comment by sudk1896** [comment]
@karlnapf: So I undo the came case changes I made and just send my test ? Should my test be with or without the camel case ?

**Comment by sudk1896** [comment]
@karlnapf: Do I create a new PR then ? And If so, what do I do about the camel case (All the tests in `SGSParseMatrix` have camel case variables).

**Comment by karlnapf** [comment]
Please read my above comments.

**Pull request closed without merging**